rbd.c

```diff
@@ -239,6 +248,110 @@ static void tcmu_rbd_service_status_update(struct tcmu_device *dev,
 
 #endif /* LIBRADOS_SUPPORTS_SERVICES */
 
+#if defined LIBRADOS_SUPPORTS_GETADDRS || defined RBD_LOCK_ACQUIRE_SUPPORT
+static void tcmu_rbd_rm_stale_entry_from_blacklist(struct tcmu_device *dev, char *addrs)
+{
+	struct tcmu_rbd_state *state = tcmur_dev_get_private(dev);
+	const char *p, *q, *end;
+	char *cmd, *addr;
+	int ret;
+
+	/*
+	 * Just skip extra chars before '[' if there has
+	 */
+	p = strchr(addrs, '[');
+	if (!p)
+		p = addrs;
+
+	/*
+	 * The addrs will a string like:
+	 * "[192.168.195.172:0/2203456141,192.168.195.172:0/4908756432]"
+	 * Or
+	 * "192.168.195.172:0/2203456141"
+	 */
+	while (1) {
+		if (p == NULL || *p == ']') {
+			return; /* we are done here */
+		} else if (*p == '[' || *p == ',') {
+			/* Skip "[" and white spaces */
+			while (*p != '\0' && !isalnum(*p)) p++;
+
+			if (*p == '\0') {
+				tcmu_dev_warn(dev, "Get an invalid address '%s'!\n", addrs);
+				return;
+			}
+
+			end = strchr(p, ',');
+			if (!end)
+				end = strchr(p, ']');
+
+			if (!end) {
+				tcmu_dev_warn(dev, "Get an invalid address '%s'!\n", addrs);
+				return;
+			}
+
+			q = end; /* The *end should be ',' or ']' */
+
+			while (!isalnum(*q)) q--;
```

Comment:
Nit: same issue here

Reply:
I removed the '\0' check here yesterday.

Because in Line#279~#286 the `strchr()` will make sure that the string between pointers `p` and `end` has no '\0'.  If there has, it will return from Line#181 or Line#290 directly.

And the `while()` here is backward decreasing the `end` pointer. And the `p` is must pointing a letter or a number, I just supposed it makes no sense to check the '\0' here.

Since you mentioned it, or maybe there has some potential issues or something that I might miss, but it is harmless to do an extra check of this. So let me add it back.